utils/src/newtype.rs

```diff
@@ -0,0 +1,31 @@
+#[macro_export]
+macro_rules! newtype {
+    ($vis:vis $name:ident($wrapped:ty)) => {
+        $vis struct $name($wrapped);
```

Comment:
As per @muursh's comment, I think the macro rule should also match a list of attibutes and expand them before the struct declaration, so we can pass in `#[derive(Blah)]` and whatnot.